docker/docker_client.go

```diff
@@ -900,13 +900,20 @@ func newBearerTokenFromHTTPResponseBody(res *http.Response) (*bearerToken, error
 // detectPropertiesHelper performs the work of detectProperties which executes
 // it at most once.
 func (c *dockerClient) detectPropertiesHelper(ctx context.Context) error {
-	// We overwrite the TLS clients `InsecureSkipVerify` only if explicitly
-	// specified by the system context
-	if c.sys != nil && c.sys.DockerInsecureSkipTLSVerify != types.OptionalBoolUndefined {
-		c.tlsClientConfig.InsecureSkipVerify = c.sys.DockerInsecureSkipTLSVerify == types.OptionalBoolTrue
+	var proxyURL *url.URL
+	if c.sys != nil {
+		proxyURL = c.sys.DockerProxyURL
+		// We overwrite the TLS clients `InsecureSkipVerify` only if explicitly
+		// specified by the system context
+		if c.sys.DockerInsecureSkipTLSVerify != types.OptionalBoolUndefined {
+			c.tlsClientConfig.InsecureSkipVerify = c.sys.DockerInsecureSkipTLSVerify == types.OptionalBoolTrue
+		}
 	}
 	tr := tlsclientconfig.NewTransport()
 	tr.TLSClientConfig = c.tlsClientConfig
+	if proxyURL != nil {
+		tr.Proxy = http.ProxyURL(proxyURL)
+	}
```

Comment:
Non-blocking: I think removing the `proxyURL` variable, and having two separate `if c.sys != nil && c.sys…` sections, would be easier to maintain over time (as the two parts of the code may be separated by more code aded in the future.)

Reply:
thanks for your suggestion~